### DISCUSSION TITLE
Can't map Application User with Domain entities.

### DISCUSSION THREAD

**Discussion by mubassir-hasan** [jasontaylordev/CleanArchitecture#655] [Q&A] [answered]
I don't have much experience in clean architecture but I am trying my best. The below code got me thinking In order to show sender's name I need to join ApplicationUser but in "Application" it doesn't have any reference. How can send Message list with sender's name
```
public class Message: BaseAuditableEntity
    {
        public long Id { get; set; }
        public string Text { get; set; }
        public string SenderId{ get; set; }
    }
 ```
Also, have another same problem need to join user table for getting assigned agent full name
```
public class Contact:AuditableEntity
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Bio { get; set; }
        public string AssignToAgentId { get; set; }
}
```

**Comment by jasontaylordev** [comment, marked as answer]
The `ApplicationUser` extends `IdentityUser` which is considered Infrastructure (relating to the Identity framework). The missing piece is a Domain entity representing your users, let's keep it simple and create a `User` in the Domain:

```csharp
public class  User
{
  public int Id { get; set; }
  public string ApplicationUserId { get; set; } // You could also just make this the primary key.

  public string Name { get; set; }
  public string Email { get; set; }
}
```

This class represents the user within your Domain. Other Domain entities, can include navigation properties to this entity. For example:

```csharp
public class  Message
{
  public int Id { get; set; }
  public string SenderId { get; set; }
  public string Text { get; set; }

  public User Sender { get; set; }
}
```

With the above approach the Domain is independent of the Identity framework. But there is some effort required to accomplish this independence.  For example, the Domain `User` and Infrastructure `ApplicationUser` will likely share some similar data, such as email. If the user updates their profile, you will need to update it in two places. Check out this PR to learn more - https://github.com/jasontaylordev/CleanArchitecture/pull/319.

There are certainly other ways to approach this problem, such as moving `ApplicationUser` into Domain and creating a dependency on the Identity framework.

The approach that you take will always depend on the technical requirements for the solution you are building. Happy Coding! 😀

**Reply by mubassir-hasan** [reply]
Thank you for your quick response. I will create and maintain a separate user class in domain  😀
